server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java

```diff
@@ -104,4 +324,274 @@ public void testSearchIdle() throws Exception {
             assertThat(resp.getHits().getTotalHits().value, equalTo(2L));
         });
     }
+
+    public void testCircuitBreakerReduceFail() throws Exception {
+        int numShards = randomIntBetween(1, 10);
+        indexSomeDocs("test", numShards, numShards*3);
```

Comment:
I love the name of this method. It makes me think "I absolutely don't care about the contents of these docs." They are just "some docs".